Scripts/RemediationScripts/Remediate-CdnFrontDoorMinTLSVersion.ps1

```diff
@@ -0,0 +1,1123 @@
+           Write-Host $([Constants]::SingleDashLine)
+           $logResource = @{}
+           $logResource.Add("ResourceGroupName",($_.ResourceGroupName))
+           $logResource.Add("ResourceName",($_.ResourceName))
+           $logResource.Add("Reason","Minimum TLS Version is set on all the custom domains of front door.")    
+           $logSkippedResources += $logResource
+        }
+        else 
+        {
+            if (-not $item.MinimumTlsVersion -ne $requiredMinTLSVersion)
+            {
+                Write-Host "Minimum TLS Version is not set on the custom domain" $item.DomainName  -ForegroundColor $([Constants]::MessageType.Warning)
+                Write-Host $([Constants]::SingleDashLine)
+            }
+            $NonCompliantFrontDoorEndpoints += $item | Select-Object @{N='ResourceGroupName';E={$item.ResourceGroupName}},
+                                                                                            @{N='ResourceName';E={$item.ResourceName}},
+                                                                                            @{N='DomainName';E={$item.DomainName}},
+                                                                                            @{N='CertificateType';E={$item.CertificateType}},
+                                                                                            @{N='MinimumTlsVersion';E={$item.MinimumTlsVersion}},
+                                                                                            @{N='isMinTLSVersionSetOnCustomDomain';E={$item.isMinTLSVersionSetOnCustomDomain}}
+        }
+    }
+    
+    $TotalNonCompliantFrontDoorEndpoints = ($NonCompliantFrontDoorEndpoints | Measure-Object).Count
+
+    if ($totalNonCompliantFrontDoorEndpoints -eq 0)
+    {
+        Write-Host "No Front Door Endpoint(s) found having minimum TLS version less than required minimum TLS version. Exiting..." -ForegroundColor $([Constants]::MessageType.Update)
+        Write-Host $([Constants]::DoubleDashLine)
+        if($AutoRemediation -and ($frontDoorsWithReqMinTLSVersion|Measure-Object).Count -gt 0) 
+        {
+            $logFile = "LogFiles\"+ $($TimeStamp) + "\log_" + $($SubscriptionId) +".json"
+            $log =  Get-content -Raw -path $logFile | ConvertFrom-Json
+            foreach($logControl in $log.ControlList){
+                if($logControl.ControlId -eq $controlIds)
+                {
+                    $logControl.RemediatedResources=$logRemediatedResources
+                    $logControl.SkippedResources=$logSkippedResources
+                }
+            }
+            $log | ConvertTo-json -depth 10  | Out-File $logFile
+        }
+        return
+    }
+
+    Write-Host "Found [$($totalNonCompliantFrontDoorEndpoints)] Endpoints of [$($totalFrontDoors)] Front Door(s) having minimum TLS version less than required minimum TLS version." -ForegroundColor $([Constants]::MessageType.Update)
```

Comment:
FrontDoors can have many endpoints, basically FrontDoor and FrontDoorEndpoint are not at same level, so this line might be confusing for users.

Reply:
Fixed